src/OrbitVulkanLayer/SubmissionTracker.h

```diff
@@ -481,17 +500,29 @@ class SubmissionTracker : public VulkanLayerProducer::CaptureStatusListener {
         device_manager_->GetPhysicalDeviceProperties(physical_device).limits.timestampPeriod;
 
     std::vector<uint32_t> query_slots_to_reset = {};
-    for (const auto& completed_submission : completed_submissions) {
+    std::vector<QueueSubmission> submissions_to_send = {};
+
+    for (auto& completed_submission : completed_submissions) {
+      bool command_buffer_queries_succeeded = QueryCommandBufferTimestamps(
+          &completed_submission, &query_slots_to_reset, device, query_pool, timestamp_period);
+      bool marker_queries_succeeded = QueryDebugMarkerTimestamps(
+          &completed_submission, &query_slots_to_reset, device, query_pool, timestamp_period);
+
+      if (command_buffer_queries_succeeded && marker_queries_succeeded) {
```

Comment:
As written offline, I am a bit concerned that this change breaks the "ordering" property of the event's we sent to the UI, i.e. vulkan events of a specific queue are ordered per submission cpu timestamp. 
I gave this another thought (what it would mean to the client and if we can work around), but I guess the best would be to keep this property on the producer/layer side. And it is actually quite easy to adjust here.
Currently, PullCompletedSubmissions returns a sorted list. So we are iterating here in order. We could simply give up as soon as we find a submission that is incomplete (i.e. add another bool here, set this in the else case, and then in the following iterations always push the submissions to the retry block).

Reply:
Yes, that seems like a simple fix. I'll change that.

Reply:
Thanks!

Reply:
Should be fixed now.